poetry.lock

```diff
@@ -646,45 +646,45 @@ files = [
 
 [[package]]
 name = "mypy"
-version = "1.2.0"
+version = "1.4.1"
```

Comment:
mypy & typing-extensions updates not needed, still cannot lint with Python 3.12 due  to issue with Pylint, will revert these updates